source/features/clone-branch.tsx

```diff
@@ -0,0 +1,107 @@
+import delegate, {DelegateEvent} from 'delegate-it';
+import gitBranch from 'octicon/git-branch.svg';
+import insertTextTextarea from 'insert-text-textarea';
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoURL, getRepoGQL} from '../libs/utils';
+
+const getBranchBaseSha = async (branchName: string): Promise<string> => {
+	const {repository} = await api.v4(`
+		repository(${getRepoGQL()}) {
+			ref(qualifiedName: "${branchName}") {
+				target {
+					oid
+				}
+			}
+		}
+	`);
+
+	return repository.ref.target.oid;
+};
+
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLButtonElement>): Promise<void> {
+	const currentTarget = event.delegateTarget;
+	const branchElement = currentTarget.closest<HTMLElement>('[data-branch-name]');
+
+	const currentBranch = getBranchBaseSha(branchElement!.dataset.branchName!);
+	let newBranchName = prompt('Enter the new branch name')?.trim();
+	if (!newBranchName) {
+		return;
+	}
+
+	let result = await createBranch(newBranchName, await currentBranch, currentTarget);
+
+	while (result !== true) {
+		newBranchName = prompt(result + '\n Enter the new branch name', newBranchName)?.trim();
+		if (!newBranchName) {
+			return;
+		}
+
+		result = await createBranch(newBranchName, await currentBranch, currentTarget); // eslint-disable-line no-await-in-loop
+	}
+
+	if (!newBranchName) {
+		return;
+	}
```

Comment:
Duplicate

Reply:
No since what if the click cancel the second time?